File: core/src/com/alta_v2/physics/utils/NpcMovementCalculator.java
package com.alta_v2.physics.utils;

import com.badlogic.gdx.Gdx;
import lombok.experimental.UtilityClass;

@UtilityClass
public class NpcMovementCalculator {

    public float getPositionX(float localX, float globalFocusX, int tileWidth) {
        float centerX = ((float) Gdx.graphics.getWidth() / 2);
        return localX * tileWidth + centerX - globalFocusX;
    }

    public float getPositionY(float localY, float globalFocusY, int tileHeight) {
        float centerY = ((float) Gdx.graphics.getHeight() / 2);
        return localY * tileHeight + centerY - globalFocusY;
    }
}

File: core/src/com/alta_v2/physics/utils/NpcMovementCalculator.kt
package com.alta_v2.physics.utils

import com.badlogic.gdx.Gdx

class NpcMovementCalculator {

    companion object {
        fun getPositionX(localX: Float, globalFocusX: Float, tileWidth: Int): Float {
            val centerX = Gdx.graphics.width.toFloat() / 2
            return localX * tileWidth + centerX - globalFocusX
        }

        fun getPositionY(localY: Float, globalFocusY: Float, tileHeight: Int): Float {
            val centerY = Gdx.graphics.height.toFloat() / 2
            return localY * tileHeight + centerY - globalFocusY
        }
    }
}